changelogs/unreleased/gh-5843-fiber-wakeup.md

```diff
@@ -0,0 +1,6 @@
+## bugfix/core
```

Comment:
If you claim #5843 is fixed, then you need to cover it with fio tests.

Reply:
As far as I understand the behavior of fio is expected due to coio engine but real bug was in assert condition as @sergos pointed. So I extracted the simple case which triggers assertion and moved it into a unit test. I think it is better.